### PR TITLE
Nuclear Operative melee class code only

### PR DESCRIPTION
**Pull request by Gannets**
## About the PR 
Finally finishes off the code for the nuclear operatives melee class crate. Not currently purchasable and no sprites yet since we're so close to the resprite launch date. I'll return to add those post-launch.

**Basic class info:**

- Two-handed sword, builds up force when attacking up to a max value. Can swap between stab and sweep special attacks, both of which apply knockback.
- Heavy armour with high protection values and stun resist. Slows the wearer considerably.
- Boots with noslip and magboot style push resist.
- Gloves with a high deflection property value.

## Why's this needed? 
Enables me to play test this class a little on live.

## Changelog

```changelog
(u)Gannets
(*)The long awaited nuclear operatives melee class is being added in a prototype state.
(*)It's listed in the syndicate weapons vendor under Knight (Prototype)
```

### CODE REVIEW

**Comment by RGBDeadSilent** [comment]
Sounds like a Nuke Op with Heavy Crusader Mk II.

Making this operative very slow should be added to both the sword and the armor. Being able to zip around with the sword could be dangerous.
Boots absolutely need no slip, wasn't said in description.
Either the sword itself, or the operative needs gloves that disables disarm. Dropping this thing basically means the operative is dead, and whoever has the sword can instantly dunk on the nuke and destroy it.
(Biolocking Nuke Op weapons would let nuke ops win so much more because the crew can't turn their weapons on them.)
Heavy saber didn't have reflect, very vulnerable to ranged stun/stolen Ops weapons. Ops can easily be taken down solo.

File: code/obj/item/misc_weapons.dm [hunk 1]
```diff
@@ -1471,21 +1473,30 @@ obj/item/whetstone
 	dropped(mob/user)
 		..()
 		if (isturf(src.loc))
-			user.visible_message("<span class='alert'>[src] falls from [user]'s hands and powers down!</span>")
-			force = 50
+			user.visible_message("<span class='alert'>[src] fdrops from [user]'s hands and powers down!</span>")
+			force = 25
```
Comment (Azrun) [inline code/obj/item/misc_weapons.dm:1477]:
```suggestion
			force = initial(src.force)
```
Not necessary, just improves maintainability by reducing magic numbers.

File: code/obj/item/misc_weapons.dm [hunk 2]
```diff
@@ -1471,21 +1473,30 @@ obj/item/whetstone
...
+				src.mode = 2
+				user.update_inhands()
+				src.setItemSpecial(/datum/item_special/rangestab)
+			if(2)
+				boutput(user, "<span class='alert'>[src] transforms in order to swing wide!</span>")
+				icon_state = "hadar_sword2"
+				item_state = "hadar_sword2"
+				src.mode = 1
+				user.update_inhands()
+				src.setItemSpecial(/datum/item_special/swipe)
```
Comment (Azrun) [inline code/obj/item/misc_weapons.dm:1495]:
```suggestion
				src.setItemSpecial(/datum/item_special/rangestab)
				tooltip_rebuild = TRUE
			if(2)
				boutput(user, "<span class='alert'>[src] transforms in order to swing wide!</span>")
				icon_state = "hadar_sword2"
				item_state = "hadar_sword2"
				src.mode = 1
				user.update_inhands()
				src.setItemSpecial(/datum/item_special/swipe)
				tooltip_rebuild = TRUE
```
Rebuild tooltip so special stats can be seen on mouseover.

File: code/obj/item/clothing/shoes.dm [hunk 3]
```diff
@@ -446,6 +446,12 @@
 			. = "Looks like some big shoes to fill!"
 		. = ..()
 
+/obj/item/clothing/shoes/swat/superheavy // so heavy you can't get shoved!
+	name = "hi-grip assault boots"
+	desc = "Specialist combat boots designed to provide enhanced grip and ankle stability."
+	icon_state = "swatheavy"
+	magnetic = 1
```
Comment (Azrun) [inline code/obj/item/clothing/shoes.dm:453]: Magnetic only seems to impact push/pulling.  

Disarm/Shove appear to disregard this, deflection property can circumvent those but only up to 50% chance of ignoring it.

**Comment by Azrun** [comment]
Design Comments:
Hadar having w_class of 4 and two handed seems like a HUGE hamper, as you throw away the sword if you want to do anything with your inventory. Having a mode where it switches to 1 handed and/or converting to w_class 3 seems like it would open up things unless they are supposed rely entirely on other operatives. (Behavior like slamgun)

Ear/Eye Insulation on Body Armor seems kinda odd, reduces value of helm.

Spreading speed reduction across armor would also limit ability to mitigate speed reduction of slipping off the armor momentarily.  Spreading most of the stats out also mitigates crew's ability to counter the class by removing a piece of equipment.

If their hands are stuck being full this all becomes a non-issue.

**Comment by Gannets** [comment]
> Sounds like a Nuke Op with Heavy Crusader Mk II.

Yeah, that's the sword I made for this class. I wasn't aware people were spawning it before now, very much unfinished.

**Pull request merged**